src/demo/examples/visualization-bar/index.js

```diff
@@ -0,0 +1,90 @@
+
+import { visualizationBar, VisualizationBar, VisualizationBarSizes } from 'hexagon-js';
+import { div } from 'utils/selection';
+
+const randNum = max => Math.round(Math.random() * max);
+const rand = max => e => {
+  e.progress(randNum(max));
+};
+
+export default () => {
+  const selectionWithMax = div();
+  const selectionErrored = div();
+  const selectionWithoutMax = div();
+  const selectionWithContent = div();
+  const selectionWithLabels = div();
+  const selectionSmall = div();
+
+  const mocks = [
+    [new VisualizationBar(selectionWithMax.node(), {
+      title: 'With maximum value', buffer: 2, value: 5, max: 20,
+    }), rand(20)],
+    [new VisualizationBar(selectionErrored.node(), {
+      title: 'With maximum value', buffer: 5, value: 10, max: 20, withPercent: true,
+    }), el => el.valid(!el.valid())],
+    [new VisualizationBar(selectionWithoutMax.node(), {
+      title: 'Without maximum value', buffer: 2, value: 5, withPercent: true,
+    }), rand(20)],
+    [new VisualizationBar(selectionWithContent.node(), {
+      title: 'With content', buffer: 2, value: 5, withPercent: true, content: 'Hello world!',
+    }), rand(20)],
+    [new VisualizationBar(selectionWithLabels.node(), {
+      title: 'With labels content',
+      buffer: 2,
+      value: 5,
+      max: 100,
+      withPercent: true,
+      content: 'Hello world!',
+      progressLabel: 'Progress',
+      bufferLabel: 'Buffer Long Text!',
+      balanceLabel: 'Balance',
+    }), rand(1000)],
+    [new VisualizationBar(selectionSmall.node(), {
+      size: VisualizationBarSizes.SMALL,
+      title: 'Small progress bar',
```

Comment:
The `ProgressBar` and `VisualizationBar` should probably be separate components to reduce the API complexity

They do share some common visuals so could both initialise a shared class that has all the functionality of both components